launcher/launcher_test.go

```diff
@@ -800,6 +809,40 @@ func TestCanConfigureCustomSampler(t *testing.T) {
 	assert.Same(t, config.Sampler, sampler)
 }
 
+func TestCanUseCustomSampler(t *testing.T) {
+	expectedSamplerProvidedAttribute := attribute.String("test", "value")
+	sampler := &testSampler{
+		decsision: trace.RecordAndSample,
+		attributes: []attribute.KeyValue{
+			expectedSamplerProvidedAttribute,
+		},
+	}
+
+	traceServer := &dummyTraceServer{}
+	stopper := dummyGRPCListenerWithTraceServer(traceServer)
+	defer stopper()
+
+	shutdown, _ := ConfigureOpenTelemetry(
+		WithSampler(sampler),
+		withTestExporters(),
+	)
+
+	tracer := otel.GetTracerProvider().Tracer("launcher-tests")
+	_, span := tracer.Start(context.Background(), "test-span")
+	span.End()
+	shutdown()
+
+	spans := traceServer.recievedExportTraceServiceRequests[0].ResourceSpans[0].ScopeSpans[0].Spans
+	require.Equal(t, 1, len(spans), "Should only be one span")
+
+	attrs := spans[0].Attributes
+	require.Equal(t, 1, len(attrs), "Should only be one attribute")
```

Comment:
Hmm, so I used require because I wanted execution to stop at that point (because otherwise it'll panic on the next line when we try to access a element that doesn't exist).

Would you prefer this to use `require` or something like `if !assert.something { t.Fail() }`?

Reply:
Yeah, I think using require is the correct thing here. Not sure why I that went over my head yesterday.

Reply:
No worries, thanks for the review :D